seed/static/seed/js/controllers/show_populated_columns_modal_controller.js

```diff
@@ -17,116 +17,19 @@ angular.module('SEED.controller.show_populated_columns_modal', []).controller('s
-
-      const relatedCols = _.filter($scope.columns, 'related');
-      // console.log('relatedCols', relatedCols);
-
-      const col_key = provided_inventory ? 'column_name' : 'name';
-
-      _.forEach(inventory, (record, index) => {
-        // console.log(cols.length + ' remaining cols to check');
-        _.forEachRight(cols, (col, colIndex) => {
-          if (notEmpty(record[col[col_key]])) {
-            // console.log('Removing ' + col[col_key] + ' from cols');
-            cols.splice(colIndex, 1);
-          }
-        });
-
-        _.forEach(record.related, (relatedRecord) => {
-          // console.log(relatedCols.length + ' remaining related cols to check');
-          _.forEachRight(relatedCols, (col, colIndex) => {
-            if (notEmpty(relatedRecord[col[col_key]])) {
-              // console.log('Removing ' + col[col_key] + ' from relatedCols');
-              relatedCols.splice(colIndex, 1);
-            }
-          });
-        });
-
-        $scope.progress = (index / inventory.length) * 50 + 50;
-      });
-
-      // determine hidden columns
-      const visible = _.reject($scope.columns, (col) => {
-        if (!col.related) {
-          return _.find(cols, { id: col.id });
-        }
-        return _.find(relatedCols, { id: col.id });
-      });
-
-      const hidden = _.reject($scope.columns, (col) => _.find(visible, { id: col.id }));
-
-      _.forEach(hidden, (col) => {
-        col.visible = false;
-      });
-
-      const columns = [];
-      _.forEach(visible, (col) => {
-        columns.push({
-          column_name: col.column_name,
-          id: col.id,
-          order: columns.length + 1,
-          pinned: col.pinnedLeft,
-          table_name: col.table_name
-        });
-      });
+    $scope.start = () => {
+      $scope.state = 'running';
+      $scope.status = 'Processing...';
+      $scope.inventory_type = inventory_type === 'properties' ? 'Property' : 'Tax lot';
 
-      const { id } = $scope.currentProfile;
-      const profile = _.omit($scope.currentProfile, 'id');
-      profile.columns = columns;
-      inventory_service.update_column_list_profile(id, profile).then((/* updatedProfile */) => {
+      inventory_service.update_column_list_profile_to_show_populated(currentProfile.id, cycle.id, $scope.inventory_type).then((/* updatedProfile */) => {
         modified_service.resetModified();
         $scope.progress = 100;
         $scope.state = 'done';
-        $scope.status = `Found ${visible.length} populated columns`;
+        $scope.refresh();
       });
     };
 
-    $scope.start = () => {
-      $scope.state = 'running';
-      $scope.status = 'Fetching Inventory';
-
-      if (provided_inventory) {
-        update_profile_with_populated_columns(provided_inventory);
-      } else {
-        const page = 1;
-        const chunk = 5000;
-        fetch(page, chunk).then(update_profile_with_populated_columns);
-      }
-    };
-
     $scope.refresh = () => {
       spinner_utility.show();
```

Comment:
```suggestion
      spinner_utility.show();
      $uibModalInstance.close();
```
Might be nice to close the modal once refresh is initiated